src/modules/products/controllers/search.controller.ts

```
@@ -1,53 +1,68 @@
+        );
+      }
+      this.logger.log(`Cache miss for key: ${cacheKey}, storing result`);
+      return result;
 
-      return this.searchService.searchProducts(search, filter);
-    } catch (error) {
-      this.logger.error("Error in searchProducts", error);
-      throw error;
+    } catch (cacheError) {
+      this.logger.warn(
+        `Failed to set cache for key: cacheKey, reason: ${cacheError.message}`
```

ESLint

[error] Insert `,`

(prettier/prettier)